electron/launcher/main-process/helpers/login.js

```diff
@@ -0,0 +1,203 @@
+        else if(cookieObject.hasOwnProperty('sessionid')){
+          sessionKey = cookieObject.sessionid;  // GLOBAL VARIABLE
+          promises.push(keytar.setPassword('Dietstory_Session', body.name, cookieObject.sessionid));
+          expiry = Date.parse(cookieObject.expires);
+        }
+      }
+      
+      return Promise.all(promises).then(() => {
+        return expiry;
+      })
+      .catch(error => {
+        throw error;
+      });
+    })
+    // SAVE USER INFO FOR FUTURE AUTO-LOGIN TO USE (ENCRYPTED)
+    .then(expiry => {
+      let encrypt = string => {
+        return new Promise((resolve, reject) => {
+          let cipher = crypto.createCipheriv(secret.algorithm, Buffer.from(secret.key, 'hex'), Buffer.from(secret.iv, 'hex'));
+          let encrypted = cipher.update(string, 'utf8', 'hex');
+          encrypted += cipher.final('hex');
+          resolve(encrypted);
+        });
+      }
+
+      sessionExpiry = expiry;   // GLOBAL VARIABLE
+      let userInfo = {...body, session_expiry: expiry};
+
+      return encrypt(JSON.stringify(userInfo)).then(encyrptedData => {
+        fs.writeFileSync(path.join(app.getPath('userData'), 'user_info'), encyrptedData, 'utf-8');
+        return userInfo;
+      })
+      .catch(error => {
+        throw error;
+      });
+    })
+    // RETURNS DJANGO RESPONSE BODY
+    .then(userInfo => {
+      resolve({body: body, userInfo: userInfo});
+    })
+    .catch(error => {
+      reject(error);
+    });
+  });
+}
+
+
+
+
+
+
+// AUTO LOGIN
+const autoLogin = () => {
+  return new Promise((resolve, reject) => {
+    // READ AND DECRYPT user_info FILE
+    fs.readFile(path.join(app.getPath('userData'), 'user_info'), 'utf-8').then(encryptedData => {
+      let decrypt = data => {
+        return new Promise((resolve, reject) => {
+          let encrypted = Buffer.from(data, 'hex');
+          let decipher = crypto.createDecipheriv(secret.algorithm, Buffer.from(secret.key, 'hex'), Buffer.from(secret.iv, 'hex'));
+          let decrypted = decipher.update(encrypted);
+          decrypted = Buffer.concat([decrypted, decipher.final()]);
+          resolve(decrypted.toString());
+        })
+      }
+
+      return decrypt(encryptedData).then(userInfo => {
+        return JSON.parse(userInfo);
+      })
+    })
+    // GET USER'S PASSWORD
+    .then(userInfo => {
+      return keytar.getPassword('Dietstory', userInfo.name).then(password => {
+        return aes256.decrypt(password).then(decryptedPassword => {
+          return {...userInfo, password: decryptedPassword};
+        })
+      })
+      .catch(error => {
+        throw error;
+      });
+    })
+    // GET SESSION TOKEN
+    .then(userInfo => {
+      // SESSION IS EXPIRED, MUST RE-LOG
+      if(Date.now() > userInfo.session_expiry){
+        return login({username: userInfo.name, password: userInfo.password}).then(response => {
+          return keytar.getPassword('Dietstory_Session', response.userInfo.name).then(sessionToken => {
```

Comment:
where is `userInfo.password` coming from? Shouldn't we be using `decryptedPassword`?

Reply:
Take a look at the previous `.then()` block where `// GET USER'S PASSWORD`. When the user_info file is decrypted, it returns an object that is the same as `userInfo` but with the added property of `password`.